main/src/main/scala/sbt/Defaults.scala

```diff
@@ -342,7 +356,7 @@ object Defaults extends BuildCommon {
     sourceDirectories := Classpaths
       .concatSettings(unmanagedSourceDirectories, managedSourceDirectories)
       .value,
-    sources := Classpaths.concat(unmanagedSources, managedSources).value
+    sources := Classpaths.concat(unmanagedSources, managedSources).value.distinct
```

Comment:
is there no way to avoid this?

Reply:
reason I ask is `distinct` is like null-checks: you could fill your codebase with them. so I'd rather avoid it, and fix the root cause - if possible.

Reply:
Looking more at this file, it could be `sources := Classpaths.concatDistinct(unmanagedSources, managedSources.value` (probably not what you mean though...). I can't really think of a way to guarantee that there is no overlap between managed sources and unmanaged sources since there is nothing that enforces that these files live in isolated directories. I guess you could do something like
```scala
sources := {
  val unmanaged = unmanagedSources.value
  val unmangedDirectories = unmanaged.view.map(_.toPath.getParent).distinct
  unmanaged ++ managedSources.value.filterNot(f => unmanagedDirectories.contains(p.toPath.getParent)
}
```

Reply:
OK.. Thanks for spending the time to look into this. Let's use `Classpaths.concatDistinct` I guess.